tests/semantic_analyser.cpp

```diff
@@ -2214,6 +2214,8 @@ class semantic_analyser_btf : public test_btf
 {
 };
 
+#ifdef HAVE_BCC_KFUNC
```

Comment:
do we really need these guards? I thought we had some mocking in place. Was hoping the 'unit' test suite could be independent of syslibs/kernel features. It wasn't failing in the CI right?

Reply:
Agreed, in general unit tests should not rely on environment at all and instead use mocked data. From looking at the semantic analyser tests, I think this should already be mocked out.

Reply:
This guard checks if BCC is able to attach to kfuncs because the same guard is used in our semantic analyser [here](https://github.com/iovisor/bpftrace/blob/339ff4b116ad63f7cd546e47a6a71d1425e362de/src/ast/semantic_analyser.cpp#L2580-L2584). If it is not defined, then these tests fail (which also happens in CI).

Reply:
Ah, makes sense then.